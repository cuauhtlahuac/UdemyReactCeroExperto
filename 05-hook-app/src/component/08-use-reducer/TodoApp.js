import React, { useReducer, useEffect } from 'react';

import { useForm } from 'hooks/useForm';
import { todoReducer } from 'reducers/todoReducer';

import './todoAppStyles.css';

const init = () => {
	return [
		{
			id: new Date().getTime(),
			task: 'Learn React',
			done: false,
		},
	];
};

const TodoApp = () => {
	const [ todos, dispatch ] = useReducer(todoReducer, [], init);

	const [ { description }, handleInputChange, reset ] = useForm({
		description: '',
	});

	useEffect(
		() => {
			localStorage.setItem('todos', JSON.stringify(todos));
		},
		[ todos ],
	);

	const handleSubmit = e => {
		e.preventDefault();

		if (description.trim().length <= 1) return;

		const newTodo = {
			id: new Date().getTime(),
			task: description,
			done: false,
		};

		const action = { type: 'addTodo', payload: newTodo };

		dispatch(action);
		reset();
	};

	return (
		<div>
			<h1>
				Todo App
				<small> ( {todos.length} )</small>
			</h1>
			<hr />

			<div className="row">
				<div className="col-7">
					<ul className="list-group">
						{todos.map((todo, i) => (
							<li key={todo.id} className="list-group-item">
								<p>{`${i + 1} .- ${todo.task}`}</p>
								<button className="btn btn-danger">Borrar</button>
							</li>
						))}
					</ul>
				</div>
				<div className="col-5">
					<h4>Agregar todo</h4>
					<hr />

					<form onSubmit={handleSubmit}>
						<input
							className="form-control"
							value={description}
							type="text"
							name="description"
							placeholder="Aprender..."
							autoComplete="off"
							onChange={handleInputChange}
						/>

						<button className="btn btn-primary mt-3 btn-block" type="submit">
							Agregar
						</button>
					</form>
				</div>
			</div>
		</div>
	);
};

export default TodoApp;
