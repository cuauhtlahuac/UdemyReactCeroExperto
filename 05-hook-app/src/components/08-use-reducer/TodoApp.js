import React, { useReducer, useEffect } from 'react';

import { todoReducer } from 'reducers/todoReducer';

import TodoList from './TodoList';
import TodoForm from './TodoForm';

import './todoAppStyles.css';

const init = () => {
	return JSON.parse(localStorage.getItem('todos')) || [];
};

const TodoApp = () => {
	const [ todos, dispatch ] = useReducer(todoReducer, [], init);

	useEffect(
		() => {
			localStorage.setItem('todos', JSON.stringify(todos));
		},
		[ todos ],
	);

	return (
		<div>
			<h1>
				Todo App
				<small> ( {todos.length} )</small>
			</h1>

			<hr />

			<div className="row">
				<div className="col-7">
					<TodoList todos={todos} dispatch={dispatch} />
				</div>

				<div className="col-5">
					<TodoForm todos={todos} dispatch={dispatch} />
				</div>
			</div>

		</div>
	);
};

export default TodoApp;
