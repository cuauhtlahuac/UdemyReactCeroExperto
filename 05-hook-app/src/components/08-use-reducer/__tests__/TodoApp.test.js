import React from 'react';
import { mount } from 'enzyme';
import TodoApp from '../TodoApp';


describe('TodoApp Test', () => {
  const component = mount(<TodoApp />)
  test('should render correctly', () => {
    expect(component).toMatchSnapshot();
  })
  

  
})
