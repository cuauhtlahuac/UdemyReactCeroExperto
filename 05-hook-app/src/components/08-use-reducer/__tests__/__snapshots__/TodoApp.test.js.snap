// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp Test should render correctly 1`] = `
<div>
  <h1>
    Todo App
    <small>
       ( 
      0
       )
    </small>
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <ul />
    </div>
    <div
      className="col-5"
    >
      Array [
        <h4>
          Agregar todo
        </h4>,
        <hr />,
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="form-control"
            name="description"
            onChange={[Function]}
            placeholder="Aprender..."
            type="text"
            value=""
          />
          <button
            className="btn btn-primary mt-3 btn-block"
            type="submit"
          >
            Agregar
          </button>
        </form>,
      ]
    </div>
  </div>
</div>
`;
